### PR TITLE
feat: made deviceId collection configurable and de-coupled anonymousId and deviceId

### PR DESCRIPTION
**Pull request by desusai7**
**Made device id collection by the SDK configurable and also stopped using deviceId as anonymous Id**
  

-   Added a new API `withCollectDeviceId` in the `RSConfigBuilder` class and when this API is set to false,  SDK will not use the value of Device Id in any possible manner and you will observe the following changes:
    1.  `context.device.id` will not be sent as part of the event payload 
    2.  If the value of `anonymousId` currently used by the SDK is equal to `IDFV`, it will changed to a UUID and this would be reflected at `context.traits.anonymousId` and `.anonymousId`
   
- **Info**: Till now, SDK would use `deviceId` as `anonymousId` by default, if the user is not overriding the default value by using the `putAnonymousId` API.   
- For the Customers who are not overriding the default value of `anonymousId`, which means when the value of `anonymousId` is equal to `IDFV`, upgrading to `1.19.0` and calling `withCollectDeviceId: NO` would change the `anonymousId` of all of your existing users to a new UUID. 
- Both the above changes are introduced to make the SDK more compliant with all the compliance policies around the collection of Device Id
- Added a new `reset` API which would accept a boolean, if this API is called with `YES` the anonymousId would also be refreshed/reset with a new UUID.

### CODE REVIEW

File: Sources/Classes/Headers/Public/RSClient.h [hunk 1]
```diff
@@ -59,7 +56,8 @@ NS_ASSUME_NONNULL_BEGIN
 - (void) identify:(NSString *_Nullable)userId traits:(NSDictionary<NSString*, id>*)traits;
 - (void) identify:(NSString *_Nullable)userId;
 
-- (void)reset;
+- (void)reset __attribute((deprecated("Discontinuing support. Use [[RSClient sharedInstance] reset: false] instead")));
```
Comment (pallabmaiti) [inline Sources/Classes/Headers/Public/RSClient.h:59]:
```suggestion
- (void)reset __attribute((deprecated("Discontinuing support. Implement reset: instead"")));
```

**Comment by pallabmaiti** [comment]
Test case failing.

File: Sources/Classes/Headers/Public/RSClient.h (hunk 1 above)
Comment (desusai7) [reply Sources/Classes/Headers/Public/RSClient.h]: Updated

**Review by pallabmaiti — COMMENTED** [review]
SonarCloud analysis failed.

**Review by pallabmaiti — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**